### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by karthi07**
Tic Tac Toe project Milestone 3

 - Create the game winning logic and game tie logic
 - added new file for game board class 
 - print and get input is only at main.rb file

### CODE REVIEW

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job thus far! 🙌 Please see my comments. 👇 
* [ ] The project uses at least two classes (Usually it uses three: Game, Board and Player). The code of the project should be logically divided between classes. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/classes.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [ ] All classes should be kept in `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

**Kindly submit for another code review request once these issues have been fixed.** _Happy coding!_ 🎉

**Review by pelzolga123 — CHANGES_REQUESTED** [review]
The project requires changes.

**Comment by pelzolga123** [comment]
## Hi Team!  
### Your project looks good :+1: 
But still, there are more changes required, before this Pull Request can be approved:
* [x] The game allows users to live the name blank. It would be great if you could add some validation to prevent it.
* [x] The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing). Your `game_play` is a long method that is doing multiple steps of the game process. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/single_responsibility.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [x] The game is behaving weirdly after the player entered invalid move and correct one after that:
image

### Please kindly fix the above issues and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

File: bin/main.rb [hunk 1]
```diff
@@ -1,3 +1,116 @@
...
+    while player2.length.zero?
+      puts "Name cant be blank. Enter a valid name \n"
+      player2 = gets.chomp
+    end
+    puts ' '
+    puts player2 + ' will be playing with O'
+    [player1, player2]
+  end
+
+  def game_play() # rubocop:disable Metrics/MethodLength
```
Comment (mcrd25) [inline bin/main.rb:77]: [OPTIONAL]
As mentioned before, your method should follow the single responsibility rule. 
I congratulate you on your attempt so far! 👏 
There are still however some components of its method that you can improve. For instance lines, 79 to 82 could be a method called set players etc.

**Review by mcrd25 — APPROVED** [review]
# Changes Approved 👍 💪 🎉  ㊗️ 

Hey @karthi07! Hey @suretrust! 👋 

Great job with changes!
No need for further review on my end! 😄 

***If you have any questions, now or in the future, do not hesitate to contact me on social media, email or on slack and I will be glad to chat with you.***

*Happy coding 😄,*
**[--Maya Douglas](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

**Pull request merged**